src/Network/GRPC/MQTT/Message/Packet.hs

```diff
@@ -158,55 +164,110 @@ makePacketReader queue = do
 
 -- Packet - Packet Senders -----------------------------------------------------
 
+newtype PacketSizeError = PacketSizeError
+  {getPacketSizeError :: Integer}
+  deriving (Data, Eq, Generic, Ord, Show, Typeable)
+
+-- | @since 1.0.0
+instance Exception PacketSizeError where
+  displayException (PacketSizeError n) =
+    Text.printf
+      "%s: maximum packet payload size must be between %d and %d bytes."
+      (show (PacketSizeError n) :: String)
+      minPacketSize
+      maxPacketSize
+  {-# INLINE displayException #-}
+
+-- | The predicate 'isValidPacketSize' tests if the 'Word32' can be used as the
+-- maximum packet payload size in 'makePacketSender'. This is any 'Word32' value
+-- that is:
+--
+-- * Greater than or equal to 'minPacketSize'.
+--
+-- * Strictly less than 'maxPacketSize'.
+--
+-- @since 1.0.0
+isValidPacketSize :: Word32 -> Bool
+isValidPacketSize x = minPacketSize <= x && x < maxPacketSize
+
+-- | The minimum packet payload size, 16 bytes.
+--
+-- @since 1.0.0
+minPacketSize :: Word32
+minPacketSize = 16
+
+-- | The maximum packet payload size, 256mB.
+--
+-- @since 1.0.0
+maxPacketSize :: Word32
+maxPacketSize = 256 * 2 ^ (20 :: Int)
```

Comment:
Now that there is a maximum defined, I think it would make sense to use `maxPacketSize` as the default value for `mqttMsgSizeLimit` in `defaultMGConfig`.

Reply:
Unless theres a specific reason for why you feel like it makes sense, I think it would be best to leave it as is. The default message size is within the new bounds and is (approximately) the midpoint between the lower and upper bound.

Reply:
I guess I don't feel too strongly about it. It just seemed arbitrary to default to the limit of AWS's implementation. I'm not sure if such a limitation is common among other MQTT brokers.

Reply:
Generally speaking even AMQP recommends setting a max message size that is similar to that set by AWS' MQTT broker. I'm not entirely sure what the rationale is.

However, I could imagine a scenario where we chose to adjust the max message size rabbitmq supported and therefore adjust the max message size in grpc-mqtt. So I'm in-favor of setting it based on a single config parameter.